specs/messengers.md

```diff
@@ -27,20 +27,35 @@ upgrade.
 The `L2CrossDomainMessenger` is a predeploy contract located at
 `0x4200000000000000000000000000000000000007`.
 
+The base `CrossDomainMessenger` interface is:
+
 ```solidity
 interface CrossDomainMessenger {
     event FailedRelayedMessage(bytes32 indexed msgHash);
     event RelayedMessage(bytes32 indexed msgHash);
     event SentMessage(address indexed target, address sender, bytes message, uint256 messageNonce, uint256 gasLimit);
-
-    function MESSAGE_VERSION() view external returns (uint16);
-    function messageNonce() view external returns (uint256);
-    function otherMessenger() view external returns (address);
-    function failedMessages(bytes32) view external returns (bool);
-    function relayMessage(uint256 _nonce, address _sender, address _target, uint256 _value, uint256 _minGasLimit, bytes memory _message) payable external;
-    function sendMessage(address _target, bytes memory _message, uint32 _minGasLimit) payable external;
-    function successfulMessages(bytes32) view external returns (bool);
-    function xDomainMessageSender() view external returns (address);
+    event SentMessageExtension1(address indexed sender, uint256 value);
+
+    function MESSAGE_VERSION() external view returns (uint16);
+    function MIN_GAS_CALLDATA_OVERHEAD() external view returns (uint64);
+    function MIN_GAS_CONSTANT_OVERHEAD() external view returns (uint64);
+    function MIN_GAS_DYNAMIC_OVERHEAD_DENOMINATOR() external view returns (uint64);
+    function MIN_GAS_DYNAMIC_OVERHEAD_NUMERATOR() external view returns (uint64);
```

Comment:
Not sure if these should be included tbh.

Reply:
I'd like to autogen the interfaces because they fall out of date easily and include links to them here. We don't need to do this now

Reply:
Totally. Until we get that in place, I think we should just have a placeholder like "TODO: Add interface here", and then fix that up during a polish before deploying.